### ISSUE TITLE
Audio analysis: extract timbre from reference track and generate chain

### ISSUE THREAD

**Issue by jpfaria** [jpfaria/OpenRig#164]
## Description

Analyze a reference audio track (MP3/WAV or streaming URL) to extract guitar timbre characteristics and automatically generate an OpenRig chain that reproduces the tone.

### Flow

```
User provides audio (file, URL, or Spotify/YouTube link)
    ↓
Source separation — isolate guitar from the mix
    ↓
Feature extraction — analyze timbre characteristics
    ↓
Parameter mapping — match features to plugin parameters
    ↓
Chain generation — build YAML using available plugins (#162, #163)
```

### Phase 1: Source Separation

Isolate the guitar track from a full mix using open-source models:

| Tool | Description |
|------|-------------|
| [Demucs](https://github.com/facebookresearch/demucs) (Meta) | State-of-the-art source separation, splits into drums/bass/vocals/other |
| [Open-Unmix](https://github.com/sigsep/open-unmix-pytorch) | Lighter alternative for instrument separation |
| [Spleeter](https://github.com/deezer/spleeter) (Deezer) | Fast 2/4/5 stem separation |

The "other" stem from Demucs typically contains guitar, keys, and synths.

### Phase 2: Feature Extraction

Analyze the isolated guitar audio to extract timbre features:

| Feature | What it tells us | Maps to |
|---------|-----------------|---------|
| **Spectral centroid** | Brightness — high = bright, low = warm | Treble/tone knobs |
| **Harmonic ratio** | Clean vs distorted — low ratio = more distortion | Gain/drive amount |
| **MFCC profile** | Tonal fingerprint — unique per amp/cab combo | Amp + cab model selection |
| **Spectral rolloff** | High-frequency cutoff — cab/speaker character | Cab model, tone knob |
| **RMS envelope** | Attack/sustain/decay — compression character | Compressor settings |
| **Reverb RT60** | Reverb tail length | Reverb room_size, decay |
| **Echo detection** | Delay time and feedback amount | Delay time_ms, feedback |
| **Modulation rate** | Chorus/tremolo/vibrato speed | Modulation rate_hz |
| **Stereo width** | Mono vs wide stereo | Chorus spread, delay L/R |

Tools:
- **[Essentia](https://essentia.upf.edu/)** (C++) — production-grade audio feature extraction
- **[librosa](https://librosa.org/)** (Python) — MFCCs, spectral features, onset detection
- **[aubio](https://aubio.org/)** (C) — lightweight real-time audio analysis

### Phase 3: Parameter Mapping

Convert extracted features to plugin parameters:

```
spectral_centroid: 3500 Hz (bright)
    → treble: 70%, presence: 60%

harmonic_ratio: 0.3 (heavy distortion)
    → gain pedal drive: 20% (boost into amp)
    → amp gain: 80%

mfcc_profile: matches "Mesa Rectifier" profile
    → amp model: nam_mesa_rectifier

spectral_rolloff: 4000 Hz (dark cab)
    → cab model: marshall_4x12_v30

reverb_rt60: 1.2s
    → reverb: plate, room_size: 50%, mix: 25%

echo_time: 350ms, echo_feedback: 0.4
    → delay: time_ms: 350, feedback: 40%, mix: 30%
```

### Phase 4: Reference Library

Build a database of analyzed tones from known recordings:
- Analyze reference tracks of famous tones (Master of Puppets, Comfortably Numb, etc.)
- Store feature vectors as "tone fingerprints"
- When user provides new audio, find closest match in library
- Use matched tone as starting point, fine-tune with extracted features

### Integration with OpenRig

- **CLI mode**: `openrig analyze --file guitar_sample.wav --output chain.yaml`
- **GUI mode**: "Analyze Tone" button — drag audio file, generates chain preview
- **Live mode**: analyze audio input in real-time, suggest closest preset

### Dependencies

- #162 (Plugin catalog — know what's available)
- #163 (AI timbre generator — chain generation from descriptions)
- Source separation model (Demucs recommended)
- Audio analysis library (Essentia or librosa)

### Acceptance criteria

- [ ] Can separate guitar from a full mix
- [ ] Extract at least 5 timbre features from audio
- [ ] Map features to plugin parameters with reasonable accuracy
- [ ] Generate valid chain YAML from audio analysis
- [ ] Reference library with at least 10 analyzed famous tones

**Comment by jpfaria** [comment]
Closing: this is being delivered as a Claude Code skill instead of in-app feature — outside the OpenRig runtime scope.
